sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,469 @@
+		const veKwentaTokenContract = new EthCallContract(veKwentaToken.address, ERC20ABI);
+
+		const [
+			rewardEscrowBalance,
+			stakedNonEscrowedBalance,
+			stakedEscrowedBalance,
+			claimableBalance,
+			kwentaBalance,
+			weekCounter,
+			totalStakedBalance,
+			vKwentaBalance,
+			vKwentaAllowance,
+			kwentaAllowance,
+			epochPeriod,
+			veKwentaBalance,
+			veKwentaAllowance,
+		]: ethers.BigNumber[] = await this.sdk.context.multicallProvider.all([
+			RewardEscrowContract.balanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.nonEscrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.escrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.earned(this.sdk.context.walletAddress),
+			KwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			SupplyScheduleContract.weekCounter(),
+			KwentaStakingRewardsContract.totalSupply(),
+			vKwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			vKwentaTokenContract.allowance(this.sdk.context.walletAddress, vKwentaRedeemer.address),
+			KwentaTokenContract.allowance(this.sdk.context.walletAddress, KwentaStakingRewards.address),
+			MultipleMerkleDistributorContract.distributionEpoch(),
+			veKwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			veKwentaTokenContract.allowance(this.sdk.context.walletAddress, veKwentaRedeemer.address),
+		]);
+
+		return {
+			rewardEscrowBalance: wei(rewardEscrowBalance),
+			stakedNonEscrowedBalance: wei(stakedNonEscrowedBalance),
+			stakedEscrowedBalance: wei(stakedEscrowedBalance),
+			claimableBalance: wei(claimableBalance),
+			kwentaBalance: wei(kwentaBalance),
+			weekCounter: Number(weekCounter),
+			totalStakedBalance: wei(totalStakedBalance),
+			vKwentaBalance: wei(vKwentaBalance),
+			vKwentaAllowance: wei(vKwentaAllowance),
+			kwentaAllowance: wei(kwentaAllowance),
+			epochPeriod: Number(epochPeriod),
+			veKwentaBalance: wei(veKwentaBalance),
+			veKwentaAllowance: wei(veKwentaAllowance),
+		};
+	}
+
+	public async getEscrowData() {
+		const { RewardEscrow } = this.sdk.context.contracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const RewardEscrowContract = new EthCallContract(RewardEscrow.address, RewardEscrowABI);
+
+		const schedules = await RewardEscrow.getVestingSchedules(
+			this.sdk.context.walletAddress,
+			0,
+			1000
+		);
+
+		const vestingSchedules = schedules.filter((schedule) => schedule.escrowAmount.gt(0));
+
+		const calls = [];
+
+		for (const schedule of vestingSchedules) {
+			calls.push(
+				RewardEscrowContract.getVestingEntryClaimable(
+					this.sdk.context.walletAddress,
+					schedule.entryID
+				)
+			);
+		}
+
+		const vestingEntries: {
+			quantity: ethers.BigNumber;
+			fee: ethers.BigNumber;
+		}[] = await this.sdk.context.multicallProvider.all(calls);
+
+		const escrowData = [];
+
+		let totalVestable = 0;
+
+		for (let i = 0; i < vestingSchedules.length; i++) {
```

Comment:
`reduce` would be a cleaner method here, maybe just copied code though.

Reply:
I decided to use a for loop for the performance gains in this scenario.